lambda/translprim.ml

```diff
@@ -594,6 +596,16 @@ let comparison_primitive comparison comparison_kind =
   | Compare, Compare_int32s -> Pcompare_bints Pint32
   | Compare, Compare_int64s -> Pcompare_bints Pint64
 
+let lazy_from_val env pty arg loc =
+  match is_function_type env pty with
+  | None ->
+     Matching.lazy_from_val arg loc
+  | Some (argty, _) ->
+     if Typeopt.lazy_val_requires_forward env argty then
+       Matching.lazy_from_val arg loc
+     else
+       arg
```

Comment:
Why do that in `lambda_of_prim`. Wouldn't it fit better in `specialize_prim` ? You would not have to add the environment to lambda_of_prim.